### PR TITLE
Neat 279 add write endpoint

### PR DESCRIPTION
**Pull request by nikokaoja**
### Improved
- Single `NeatGraphStore` instantiated via three options:
  - `from_memory_store`
  - `from_oxi_store`
  - `from_sparql_store`
### Removed
- Removed various superclassing of `NeatGraphStore`
- Remove Prometheus reminisce in code base
- Remove logging
### Added
- `RdfFileExtractor` graph extractor added.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,20 @@ Changes are grouped as follows:
...
+## [0.80.0] - 10-06-24
+
+### Improved
+- Single `NeatGraphStore` instantiated via three options:
+  - `from_memory_store`
+  - `from_oxi_store`
+  - `from_sparql_store`
+### Removed
+- Removed various superclassing of `NeatGraphStore`
+- Remove Prometheus reminisce in code base
```
Comment (doctrino) [inline docs/CHANGELOG.md:28]: Oh yes

File: cognite/neat/graph/stores/_base.py [hunk 2]
```diff
@@ -1,291 +1,165 @@
...
-            self.graph_name,
-            self.base_prefix,
-            self.returnFormat,
-            self.internal_storage_dir,
-        )
+        self.queries = _Queries(self)
 
-    def upsert_prefixes(self, prefixes: dict[str, Namespace]) -> None:
+    @classmethod
+    def _upsert_prefixes(cls, provenance: Provenance, graph: Graph, rules: InformationRules) -> None:
```
Comment (doctrino) [inline cognite/neat/graph/stores/_base.py:58]: Any reason not to just make this a regular method?
```suggestion
    def _upsert_prefixes(self, prefixes: Iterable[Namespace]):-> None:
```

File: cognite/neat/graph/stores/_base.py [hunk 3]
```diff
@@ -1,291 +1,165 @@
...
-        self.rdf_store_query_url = rdf_store_query_url
-        self.rdf_store_update_url = rdf_store_update_url
-        self.graph_name = graph_name
-        self.returnFormat = returnFormat
-        self.internal_storage_dir = Path(internal_storage_dir) if internal_storage_dir else None
-        self.internal_storage_dir_orig = (
-            self.internal_storage_dir if self.internal_storage_dir_orig is None else self.internal_storage_dir_orig
-        )
+        if self.rules:
+            self._upsert_prefixes(self.provenance, self.graph, self.rules)
```
Comment (doctrino) [inline cognite/neat/graph/stores/_base.py:53]: If you do the comment below, then this becomes 
```suggestion
            self._upsert_prefixes(self.rules.prefixes.values())
```
which is more readable to me

File: cognite/neat/graph/stores/_base.py [hunk 4]
```diff
@@ -1,291 +1,165 @@
...
+            add_base_iri : Add base IRI to graph, by default True
         """
-        return _DelayedQuery(self.graph, query)
 
-    @abstractmethod
-    def drop(self) -> None:
-        """Drops the graph."""
-        raise NotImplementedError()
+        # Oxigraph store, do not want to type hint this as it is an optional dependency
+        if type(self.graph.store).__name__ == "OxigraphStore":
```
Comment (doctrino) [inline cognite/neat/graph/stores/_base.py:142]: I guess this is the trade-off for not base classing. Worth it :)

**Review by doctrino — APPROVED** [review]
Absolutely love this change! Only a minor nitpicking of switching the classmethod to a regular method. Otherwise Great!

File: cognite/neat/graph/stores/_base.py (hunk 4 above)
Comment (nikokaoja) [reply cognite/neat/graph/stores/_base.py:142]: Unft. yest

**Pull request merged**